components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/uaepass/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,958 @@
+        } catch (ParseException e) {
+            LOG.error("Error occurred while parsing user info payload by UAEPass.");
+            throw new UAEPassUserInfoFailedException("Error occurred while parsing user info payload by UAEPass.", e);
+        }
+
+        return userInfoJwtAttributes;
+    }
+
+    /**
+     * This method is used to create userinfo request with the access token.
+     *
+     * @param context                          The Authentication context received by authenticator.
+     * @param accessToken                      The access token obtained from the processAuthenticationResponse.
+     * @return String                          The response which returns from the user info API call.
+     * @throws UAEPassUserInfoFailedException  Throws an exception, if not obtains the user claims from the user info.
+     */
+    public String sendUserInfoRequest(AuthenticationContext context, String accessToken)
+            throws UAEPassUserInfoFailedException {
+
+        StringBuilder builder = new StringBuilder();
+
+        try {
+            String envUAEPass = getUAEPassEnvironment(context);
+            URL userInfoUrl = new URL(getUserInfoUrl(envUAEPass));
+            HttpURLConnection httpURLConnection = (HttpURLConnection) userInfoUrl.openConnection();
+            httpURLConnection.setRequestMethod("GET");
+            httpURLConnection.setRequestProperty("Authorization", "Bearer " + accessToken);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(httpURLConnection.getInputStream()));
+            String inputLine = reader.readLine();
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("User info request is sent successfully.");
+            }
+
+            while (inputLine != null) {
+                builder.append(inputLine).append("\n");
+                inputLine = reader.readLine();
+            }
+
+        } catch (IOException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.error("Unable to retrieve successful response from UAEPass UserInfo", e);
+            }
+            throw new UAEPassUserInfoFailedException("UAEPass UserInfo failure.", e);
+        }
+        if (LOG.isDebugEnabled() && IdentityUtil.isTokenLoggable(IdentityConstants.IdentityTokens.USER_ID_TOKEN)) {
+            LOG.debug("response: " + builder);
+        }
+
+        return builder.toString();
+    }
+
+    /**
+     * Returns the user id of the authenticated user.
+     *
+     * @param userClaims                      The Map object with user claims returns from buildJSON.
+     * @return String                         The ID of the authenticated user from UAEPass.
+     * @throws AuthenticationFailedException  Throws an AuthenticationFailedException exception to
+     *                                        processAuthenticationResponse.
+     */
+    public String getAuthenticatedUserId(Map<String, Object> userClaims)
```

Comment:
Let's check all the methods, whether we actually required to declare the method as public?

Reply:
Fixed with : [c9f22571bc5](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/c9f22571bc5ffd21428c8b3a022c61e2045de40f)